source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
`max@hood:/opt/aspect/benchmarks/vankeken_subduction$ ./aspect vankeken_subduction.prm 
-----------------------------------------------------------------------------
-- This is ASPECT, the Advanced Solver for Problems in Earth's ConvecTion.
--     . version 2.0.0-pre
--     . running in DEBUG mode
--     . running with 1 MPI process
--     . using Trilinos
-----------------------------------------------------------------------------

Loading shared library <./libprescribed_velocity_pressure.so>

Number of active cells: 7,040 (on 4 levels)
Number of degrees of freedom: 92,700 (56,994+7,209+28,497)

*** Timestep 0:  t=0 years
   Solving temperature system... 0 iterations.
------------------------------------------
Got error 2 in row 2 of proc 0 when trying to add the columns:
2 

Matrix row  has the following indices:



--------------------------------------------------------
An error occurred in line <1717> of file </opt/aspect-stack/tmp/unpack/deal.II-master/source/lac/trilinos_sparse_matrix.cc> in function
    void dealii::TrilinosWrappers::SparseMatrix::add(dealii::TrilinosWrappers::SparseMatrix::size_type, dealii::TrilinosWrappers::SparseMatrix::size_type, const size_type*, const TrilinosScalar*, bool, bool)
The violated condition was: 
    ierr <= 0
Additional information: 
    You tried to access element (2/2) of a sparse matrix, but it appears to not exist in the Trilinos sparsity pattern.

Stacktrace:
-----------
#0  /opt/aspect-stack/deal.II-master/lib/libdeal_II.g.so.9.0.0-pre: dealii::TrilinosWrappers::SparseMatrix::add(unsigned int, unsigned int, unsigned int const*, double const*, bool, bool)
#1  ./aspect: dealii::BlockMatrixBase<dealii::TrilinosWrappers::SparseMatrix>::add(unsigned int, unsigned int, double)
#2  /opt/aspect-stack/deal.II-master/lib/libdeal_II.g.so.9.0.0-pre: void dealii::ConstraintMatrix::distribute_local_to_global<dealii::TrilinosWrappers::BlockSparseMatrix, dealii::TrilinosWrappers::MPI::BlockVector>(dealii::FullMatrix<dealii::TrilinosWrappers::BlockSparseMatrix::value_type> const&, dealii::Vector<dealii::TrilinosWrappers::MPI::BlockVector::value_type> const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, dealii::TrilinosWrappers::BlockSparseMatrix&, dealii::TrilinosWrappers::MPI::BlockVector&, bool, dealii::internal::bool2type<true>) const
#3  ./aspect: aspect::Simulator<2>::copy_local_to_global_stokes_system(aspect::internal::Assembly::CopyData::StokesSystem<2> const&)
#4  ./aspect: void dealii::WorkStream::run<std::_Bind<std::_Mem_fn<void (aspect::Simulator<2>::*)(dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2, 2>, false> > const&, aspect::internal::Assembly::Scratch::StokesSystem<2>&, aspect::internal::Assembly::CopyData::StokesSystem<2>&)> (aspect::Simulator<2>*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>)>, std::_Bind<std::_Mem_fn<void (aspect::Simulator<2>::*)(aspect::internal::Assembly::CopyData::StokesSystem<2> const&)> (aspect::Simulator<2>*, std::_Placeholder<1>)>, dealii::FilteredIterator<dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2, 2>, false> > >, aspect::internal::Assembly::Scratch::StokesSystem<2>, aspect::internal::Assembly::CopyData::StokesSystem<2> >(dealii::FilteredIterator<dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2, 2>, false> > > const&, dealii::identity<dealii::FilteredIterator<dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2, 2>, false> > > >::type const&, std::_Bind<std::_Mem_fn<void (aspect::Simulator<2>::*)(dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::DoFHandler<2, 2>, false> > const&, aspect::internal::Assembly::Scratch::StokesSystem<2>&, aspect::internal::Assembly::CopyData::StokesSystem<2>&)> (aspect::Simulator<2>*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>)>, std::_Bind<std::_Mem_fn<void (aspect::Simulator<2>::*)(aspect::internal::Assembly::CopyData::StokesSystem<2> const&)> (aspect::Simulator<2>*, std::_Placeholder<1>)>, aspect::internal::Assembly::Scratch::StokesSystem<2> const&, aspect::internal::Assembly::CopyData::StokesSystem<2> const&, unsigned int, unsigned int)
#5  ./aspect: aspect::Simulator<2>::assemble_stokes_system()
#6  ./aspect: aspect::Simulator<2>::solve_timestep()
#7  ./aspect: aspect::Simulator<2>::run()
#8  ./aspect: main
--------------------------------------------------------

[hood:28671] *** Process received signal ***
[hood:28671] Signal: Aborted (6)
[hood:28671] Signal code:  (-6)
[hood:28671] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x11390)[0x7f046f414390]
[hood:28671] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x38)[0x7f046e9e3428]
[hood:28671] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x16a)[0x7f046e9e502a]
[hood:28671] [ 3] /opt/aspect-stack/deal.II-master/lib/libdeal_II.g.so.9.0.0-pre(_ZN6dealii18deal_II_exceptions9internals5abortERKNS_13ExceptionBaseEb+0x11f)[0x7f047669fc51]
[hood:28671] [ 4] /opt/aspect-stack/deal.II-master/lib/libdeal_II.g.so.9.0.0-pre(_ZN6dealii18deal_II_exceptions9internals11issue_errorINS_16TrilinosWrappers12SparseMatrix28ExcAccessToNonPresentElementEEEvNS1_17ExceptionHandlingEPKciS8_S8_S8_T_+0x36)[0x7f04765b39be]
[hood:28671] [ 5] /opt/aspect-stack/deal.II-master/lib/libdeal_II.g.so.9.0.0-pre(_ZN6dealii16TrilinosWrappers12SparseMatrix3addEjjPKjPKdbb+0xb7f)[0x7f04765a704b]
[hood:28671] [ 6] ./aspect(_ZN6dealii15BlockMatrixBaseINS_16TrilinosWrappers12SparseMatrixEE3addEjjd+0x32e)[0xe21680]
[hood:28671] [ 7] /opt/aspect-stack/deal.II-master/lib/libdeal_II.g.so.9.0.0-pre(_ZNK6dealii16ConstraintMatrix26distribute_local_to_globalINS_16TrilinosWrappers17BlockSparseMatrixENS2_3MPI11BlockVectorEEEvRKNS_10FullMatrixINT_10value_typeEEERKNS_6VectorINT0_10value_typeEEERKSt6vectorIjSaIjEERS7_RSD_bNS_8internal9bool2typeILb1EEE+0xda7)[0x7f04760cf2a5]
[hood:28671] [ 8] ./aspect(_ZN6aspect9SimulatorILi2EE34copy_local_to_global_stokes_systemERKNS_8internal8Assembly8CopyData12StokesSystemILi2EEE+0x42)[0xe26c7a]
[hood:28671] [ 9] ./aspect(_ZN6dealii10WorkStream3runISt5_BindIFSt7_Mem_fnIMN6aspect9SimulatorILi2EEEFvRKNS_18TriaActiveIteratorINS_15DoFCellAccessorINS_10DoFHandlerILi2ELi2EEELb0EEEEERNS4_8internal8Assembly7Scratch12StokesSystemILi2EEERNSG_8CopyData12StokesSystemILi2EEEEEPS6_St12_PlaceholderILi1EEST_ILi2EEST_ILi3EEEES2_IFS3_IMS6_FvRKSN_EESS_SU_EENS_16FilteredIteratorISC_EESJ_SN_EEvRKT1_RKNS_8identityIS18_E4typeET_T0_RKT2_RKT3_jj+0x345)[0xe76817]
[hood:28671] [10] ./aspect(_ZN6aspect9SimulatorILi2EE22assemble_stokes_systemEv+0x44a)[0xe777d4]
[hood:28671] [11] ./aspect(_ZN6aspect9SimulatorILi2EE14solve_timestepEv+0x326)[0xf03ddc]
[hood:28671] [12] ./aspect(_ZN6aspect9SimulatorILi2EE3runEv+0x10c3)[0xf06639]
[hood:28671] [13] ./aspect(main+0x7b4)[0xf74f22]
[hood:28671] [14] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f046e9ce830]
[hood:28671] [15] ./aspect(_start+0x29)[0x8bb249]
[hood:28671] *** End of error message ***
Aborted
max@hood:/opt/aspect/benchmarks/vankeken_subduction$ 
`